module/profiler/profiler.go

```diff
@@ -71,28 +78,52 @@ func (p *AutoProfiler) Enabled() bool {
 	return p.enabled.Load()
 }
 
-func (p *AutoProfiler) Ready() <-chan struct{} {
-	delay := time.Duration(float64(p.interval) * rand.Float64())
-	p.unit.LaunchPeriodically(p.start, p.interval, delay)
+// TriggerRun manually triggers a profile run if one is not already running.
+func (p *AutoProfiler) TriggerRun(d time.Duration) error {
+	select {
+	case p.trigger <- d:
+		return nil
+	default:
+		return errors.New("profiling is already in progress")
+	}
+}
+
+func (p *AutoProfiler) runForever() {
+	jitter := time.Duration(rand.Int63n(int64(p.interval)))
+	t := time.NewTicker(p.interval + jitter)
```

Comment:
should we recompute the jitter on each iteration?

also, the expected interval is now 1.5 * p.interval.  should we normalized?